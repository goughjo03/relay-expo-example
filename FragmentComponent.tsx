import { useFragment } from "react-relay";
// @ts-ignore
import graphql from "babel-plugin-relay/macro";
import { FragmentComponent_country$key } from "./__generated__/FragmentComponent_country.graphql";
import { Text, View } from "react-native";

export const fragment = graphql`
  fragment FragmentComponent_country on Country {
    phone
    continent {
      name
    }
  }
`;

interface Props {
  country: FragmentComponent_country$key;
}

const FragmentComponent = ({ country }: Props) => {
  const data = useFragment(fragment, country);
  return (
    <View>
      <Text>{data.continent.name}</Text>
      <Text>{data.phone}</Text>
    </View>
  );
};

export default FragmentComponent;
