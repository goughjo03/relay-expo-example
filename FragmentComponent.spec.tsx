import {
  graphql,
  RelayEnvironmentProvider,
  useLazyLoadQuery,
} from "react-relay";
import { createMockEnvironment, MockPayloadGenerator } from "relay-test-utils";
import FragmentComponent from "./FragmentComponent";
import renderer, { act } from "react-test-renderer";
import { Suspense } from "react";

describe("FragmentComponent", () => {
  it("renders correctly with automatically generated data", async () => {
    const environment = createMockEnvironment();
    const TestRenderer = () => {
      const data = useLazyLoadQuery(
        graphql`
          query FragmentComponentQuery @relay_test_operation {
            myData: country(code: "test-id") {
              # Spread the fragment you want to test here
              ...FragmentComponent_country
            }
          }
        `,
        {}
      );
      return <FragmentComponent country={(data as any).myData} />;
    };
    const tree = renderer.create(
      <RelayEnvironmentProvider environment={environment}>
        <Suspense fallback="Loading...">
          <TestRenderer />
        </Suspense>
      </RelayEnvironmentProvider>
    );
    act(() => {
      environment.mock.resolveMostRecentOperation((operation) =>
        MockPayloadGenerator.generate(operation)
      );
    });
    await new Promise((resolve) => setTimeout(resolve, 0));
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it("renders correctly with specifically generated data", async () => {
    const environment = createMockEnvironment();
    const TestRenderer = () => {
      const data = useLazyLoadQuery(
        graphql`
          query FragmentComponentTwoQuery @relay_test_operation {
            myData: country(code: "test-id") {
              # Spread the fragment you want to test here
              ...FragmentComponent_country
            }
          }
        `,
        {}
      );
      return <FragmentComponent country={(data as any).myData} />;
    };
    const tree = renderer.create(
      <RelayEnvironmentProvider environment={environment}>
        <Suspense fallback="Loading...">
          <TestRenderer />
        </Suspense>
      </RelayEnvironmentProvider>
    );
    act(() => {
      environment.mock.resolveMostRecentOperation((operation) =>
        MockPayloadGenerator.generate(operation, {
          Country() {
            return {
              phone: "123456789",
              currency: "USD",
              continent: {
                name: "Europe",
              },
            };
          },
        })
      );
    });
    await new Promise((resolve) => setTimeout(resolve, 0));
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
