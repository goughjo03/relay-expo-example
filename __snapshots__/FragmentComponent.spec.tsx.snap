// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FragmentComponent renders correctly with automatically generated data 1`] = `
<View>
  <Text>
    &lt;mock-value-for-field-"name"&gt;
  </Text>
  <Text>
    &lt;mock-value-for-field-"phone"&gt;
  </Text>
  <Text>
    &lt;mock-value-for-field-"capital"&gt;
  </Text>
  <Text />
</View>
`;

exports[`FragmentComponent renders correctly with specifically generated data 1`] = `
<View>
  <Text>
    &lt;mock-value-for-field-"name"&gt;
  </Text>
  <Text>
    123456789
  </Text>
  <Text>
    &lt;mock-value-for-field-"capital"&gt;
  </Text>
  <Text />
</View>
`;
